optimizely/optimizely_config.py

```diff
@@ -136,9 +164,144 @@ def __init__(self, project_config):
+                typed_audience.get('name'),
+                typed_audience.get('conditions')
+            )
+            optly_typed_audiences.append(optly_audience)
+
+        for old_audience in project_config.audiences:
+            # check if old_audience.id exists in new_audiences.id from typed_audiences
+            if len([new_audience for new_audience in project_config.typed_audiences
+                    if new_audience.get('id') == old_audience.get('id')]) == 0:
+                if old_audience.get('id') == "$opt_dummy_audience":
+                    continue
+                else:
+                    # Convert audiences array to OptimizelyAudience array
+                    optly_audience = OptimizelyAudience(
+                        old_audience.get('id'),
+                        old_audience.get('name'),
+                        old_audience.get('conditions')
+                    )
+                    optly_typed_audiences.append(optly_audience)
+
+        self.audiences = optly_typed_audiences
+
+    def update_experiment(self, experiment, conditions, audiences_map):
+        '''
+            Gets an OptimizelyExperiment to update, conditions from
+            the corresponding entities.Experiment and an audiences_map
+            in the form of [id:name]
+
+            Updates the OptimizelyExperiment.audiences with a string
+            of conditions and audience names.
+        '''
+        audiences = self.replace_ids_with_names(conditions, audiences_map)
+        experiment.audiences = audiences
+
+    def replace_ids_with_names(self, conditions, audiences_map):
+        '''
+            Gets conditions and audiences_map [id:name]
+
+            Returns:
+                a string of conditions with id's swapped with names
+                or None if no conditions found.
+
+        '''
+        if conditions is not None:
+            return self.stringify_conditions(conditions, audiences_map)
+        else:
+            return None
+
+    def lookup_name_from_id(self, audience_id, audiences_map):
+        '''
+            Gets and audience ID and audiences map
+
+            Returns:
+                The name corresponding to the ID
+                or None if not found
+        '''
+        name = ""
+        try:
+            name = audiences_map[audience_id]
+        except KeyError:
+            name = audience_id
+
+        return name
+
+    def stringify_conditions(self, conditions, audiences_map):
+        '''
+            Gets a list of conditions from an entities.Experiment
+            and an audiences_map [id:name]
+
+            Returns:
+                A string of conditions and names for the provided
+                list of conditions.
+        '''
+        ARGS = ConditionOperatorTypes.operators
+        condition = ""
+        ret = ""
+        length = len(conditions)
+
+        if length == 0:
+            return
+        if length == 1:
+            '''
+                Lookup ID and replace with name
```

Comment:
I think you could use one line docstring here, for the sake of space, we do multiline dostrings mainly for functions and classes